src/openms/source/ANALYSIS/OPENSWATH/MRMDecoy.cpp

```diff
@@ -184,53 +217,41 @@ namespace OpenMS
 
       ++attempts;
 
-      // If our attempts have failed so far, we will append two random AA to
+      // If our attempts have failed so far, we will mutate a random AA of
       // the sequence and see whether we can achieve sufficient shuffling with
-      // these additional AA added to the sequence.
+      // the new sequence.
       if (attempts % 10 == 9)
       {
-        if (replace_aa_instead_append)
```

Comment:
Previously, the ``shuffle`` method appended two random residues if the sequence similarity did not reach a lower threshold. However, this did not guarantee that the N-term remained constant or was even a tryptic residue, which might potentially introduce bias with the b- and y-ion series score of OpenSWATH.

For IPF, we implemented an approach to swap individual residues and the implementation here follows a similar strategy. Instead of appending, existing (non-modified and non-terminal) residues are randomly changed. A similar strategy (or ``mutated`` approach) [has been suggested](http://www.mcponline.org/content/suppl/2017/10/25/RA117.000314.DC1/4947_1_supp_14434_zy1gl9.pdf) by others as well.